workspace_tools/export/gcc_arm_common.tmpl

```diff
@@ -1,12 +1,38 @@
 # This file was automagically generated by mbed.org. For more information, 
 # see http://mbed.org/handbook/Exporting-to-GCC-ARM-Embedded
 
+ifeq ($(OS),Windows_NT)
+MKDIR := mkdir
+else
+MKDIR := mkdir -p
+endif
+
+ifeq (,$(filter bin,$(notdir $(CURDIR))))
+.SUFFIXES:
+OBJDIR := bin
+MAKETARGET = $(MAKE) --no-print-directory -C $@ -f $(CURDIR)/Makefile \
```

Comment:
that style of rules does have a drawback: any rule added to the makefile also has to have that style. On the other hand, the way I am recursively calling make has few drawbacks, as I remove all of the rules from the first invocation of make.